onnxruntime/contrib_ops/rocm/bert/tunable_op.h

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <iostream>
+#include <map>
+#include <vector>
+#include <string>
+#include "timer.h"
+
+namespace onnxruntime {
+namespace contrib {
+namespace rocm {
+
+struct OpParams {
+  virtual std::string signature() const = 0;
+};
+
+class Op {
+ public:
+  Op() : repeats_(100) {}
+
+  virtual void Run(const OpParams*) = 0;
+
+  void SetRepeats(int n) {
+    repeats_ = n;
+  }
+
+  float Profile(const OpParams* op_params) {
+    // warm up
+    for (int i = 0; i < 5; i++) {
+      Run(op_params);
+    }
+    timer_.Start();
+    for (int i = 0; i < repeats_; i++) {
+      Run(op_params);
+    }
+    timer_.End();
+    return timer_.time()/repeats_;
+  }
+
+  virtual ~Op() {}
+
+ private:
+  Timer timer_;
+  int repeats_;
+};
+
+class TunableOp {
```

Comment:
should we have an api to get the tuning result?

Reply:
This would be good extension. I'm also thinking about adding features to save and load tuning results.